### ISSUE TITLE
Choosing a language for syntax highlighting creates an extra monospace block

### ISSUE THREAD

**Issue by hyperion-cs** [telegramdesktop/tdesktop#28577]
### Steps to reproduce

1. In the message input window, enter (a minimal example):
```
x
ab c
y
```
2. Select the second line, i.e., `ab c`, then _right-click -> Formatting -> Monospace_ (or use a shortcut).
A monospace block will be generated without syntax highlighting:
<img width="706" alt="image" src="https://github.com/user-attachments/assets/d2d39682-d951-45f0-a835-4de3617f6276">

3. Click on the block's header, and in the window that appears, specify any language (e.g., `go`) for highlighting:
<img width="341" alt="image" src="https://github.com/user-attachments/assets/0e6b896a-73a7-4ee1-8c3e-75cce4a0e656">

And click _Save_.

### Expected behaviour

It is expected that we will see the same block with the same content, but with the selected language for highlighting. No additional blocks:
<img width="692" alt="image" src="https://github.com/user-attachments/assets/5e7cfe1b-f977-4a10-b1e4-c09779e7a989">

### Actual behaviour

Actually, we see that the underlying text below the block was captured, and an additional (extra) block was generated based on it.
<img width="691" alt="image" src="https://github.com/user-attachments/assets/d52e1dc7-060a-48b9-9988-b5b1fcce69ae">

This bug occurs both when creating a new message and when editing an existing one.

### Operating system

macOS, 15.0, Aqua

### Version of Telegram Desktop

5.6.3

### Installation source

Static binary from official website

**Comment by hyperion-cs** [comment]
@Aokromes @ilya-fedin could you pay attention to this issue, please?

**Comment by ilya-fedin** [comment]
I can't help, sorry, I'm mostly working on Linux-specific things and have a bad knowledge of the UI code. I'm doubt Someone's can help either, he doesn't code at all.

**Comment by hyperion-cs** [comment]
@ilya-fedin got it. Anyway, thank you for your time.

**Comment by hyperion-cs** [comment]
Still relevant.

**Comment by 23rd** [comment]
@hyperion-cs 
Do you have a Windows environment?
Would appreciate it if you could test this nightly debug build, but it is only for Windows. =(
Unfortunately, the problem is quite deep, and my fix might have affected other areas related to formatting, so it would be great if you could test the application for your various formatting-related cases.
Thanks.

https://github.com/telegramdesktop/tdesktop/actions/runs/17673001347

**Comment by hyperion-cs** [comment]
@23rd yes, I have Windows env. Should I test “Ninja Multi-Config” or the standard version?

**Comment by 23rd** [comment]
@hyperion-cs You can download any of the four builds.

**Comment by hyperion-cs** [comment]
@23rd the bug in these builds has been fixed; I haven't noticed any others. What's more, it seems that someone has fixed [this](https://github.com/telegramdesktop/tdesktop/issues/28578) bug (it's not even present in the production build), at least on Windows. Good work!

// The only thing is that when I type a fairly long message into the input window, I don't see the scroll bar (although the scroll itself works), but it seems that this is a reason for a separate issue, since it is also observed in the production build. Besides, it's not very important and doesn't interfere with work.
